docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+[Finux's Architecture](#31-architecture) section.
+
+## 3. Design
+
+---
+
+### 3.1 Architecture
+
+![ArchitectureDiagram](img/ArchitectureDiagram.png)\
+_Figure 1: **Finux** Architecture Diagram_
+
+> 💡 The diagrams are built with [PlantUML](https://plantuml.com/).
+> The `.puml` files can be found in the [diagrams](https://github.com/AY2021S2-CS2113T-W09-1/tp/tree/master/docs/diagrams) folder and
+> the generated picture diagrams are in the [img](https://github.com/AY2021S2-CS2113T-W09-1/tp/tree/master/docs/img) folder.
+
+The _**Architecture Diagram**_ above details the high-level design of the **Finux** Application.
+Brief explanations of the components involved are given below.
+
+The `Finux` component contains only one class `Finux`, It is responsible for,
+* At launch: Initializes the components in the correct sequence and attempts to load data from file.  
+* At shut down: Shuts down the components and invokes cleanup method where necessary.
+
+The rest of the Application consists of six components.
+* `Ui`: The user interface (Ui) of the App which handles all user input and Application output.
+* `Parser`: The user input parser of the CLI.
+* `CommandHandler`: The handler of parsed arguments for conversion into appropriate `Commands`.
+* `Command`: The appropriate command to be executed.
+* `RecordList`: The list of records of the Application stored in-memory.
+* `Storage`: The storage manager that handles the saving of data, or data retrieval to/from file. 
+
+Given below are sequence diagrams describing the general Application flow and how the different objects
+interact with each other.
+
+![Initialization](img/InitializationSequenceDiagram.png)\
+_Figure 2: Initialization_
+
+The initialization sequence diagram above shows the systematic flow of object creation and record
+retrieval from storage file for creation of the `RecordList` object.
+
+![Main program flow](img/CommandLooperSequenceDiagram.png)\
+_Figure 3: Main Application Loop & Exit Sequence_
+
+> 💡 The lifeline all objects should end at the destroy marker (X) but due to a limitation of PlantUML,
+> the lifeline reaches the end of diagram.
+
+This sequence diagram follows suit after initialization in _Figure 2_.\
+This shows the main flow until the `exit` command is input by the user.
+
+The following sections below will provide more details of each component.
+
+### 3.2 UI Component
+
+![UIComponentDiagram](img/UIComponentDiagram.png)\
+*Figure 4: **Ui** Class Diagram*
+
+The Ui Component consists of a `Ui` class which handles all user input and system output.
+The Ui is only dependent on the `Duke` class and does not interact directly with other classes,
+ensuring a high level of cohesiveness, low level of coupling and separation of roles.
+
+The `Ui` component actively listens for:
+* the execution of commands to print the result of a `command`
+
+### 3.3 Parser Component
+![ParserHandlerClassDiagram](img/ParserHandlerClassDiagram.png) <br>
```

Comment:
Looks a bit text-heavy, would it be possible to show only the most important methods of the 2 classes? I think it would be best to focus on the public methods, since they make up the API.